### PR TITLE
Add tests for TelemetryApiController

### PR DESCRIPTION
**Pull request by handeadiguzel**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@handeadiguzel__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: apitests/spec/device_telemetry_spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result.deviceTelemetrySummaryModel.minimumHumidity).toBeTruthy();
+                expect(result.deviceTelemetrySummaryModel.timeFrameMinutes).toBeTruthy();
+                expect(result.deviceTelemetrySummaryModel.timestamp).toBeTruthy();
+                done();
+            });
+        });
+
+        it('6. Get Map Api Key', (done) => {
+            req.get('mapApiKey', (err, resp, result: DeviceLocationData) => {
+                expect(err).not.toBeTruthy();
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: if(err || resp.status!=200)

**Comment by kvish** [comment]
other than comments above LGTM

File: apitests/spec/device_telemetry_spec.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result[0].values.humidity).toBeTruthy();
+                expect(result[0].values.temperature).toBeTruthy();
+                done();
+            });
+        });
+
+        it('5. Get Dashboard Device Pane Data', (done) => {
+            var options: request.CoreOptions = {
+                qs: {
+                    deviceId: "SampleDevice001_648"
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: use getdevice api to get a valid device and use that

File: apitests/spec/device_telemetry_spec.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        it('4. Get Device Telemetry', (done) => {
+            var options: request.CoreOptions = {
+                qs: {
+                    deviceId: "SampleDevice001_648",
+                    minTime: "2016-07-14T18:04:20.364-07:00"
+                }
+            }
+            req.get('list', options, (err, resp, result: DeviceTelemetry[]) => {
+                expect(result).toBeTruthy();
+                expect(result[0]).toBeTruthy();
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: also check for result.length >0

File: apitests/spec/device_telemetry_spec.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result.deviceLocationList[0].longitude).toBeTruthy();
+                done();
+            });
+        });
+
+        it('3. Get Device Telemetry Summary', (done) => {
+            var options: request.CoreOptions = {
+                qs: {
+                    deviceId: "SampleDevice001_648"
+                }
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: same as others, avoid hardcoding device id

File: apitests/spec/device_telemetry_spec.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result.timeFrameMinutes).toBeTruthy();
+                expect(result.timestamp).toBeTruthy();
+                done();
+            });
+        });
+
+        it('4. Get Device Telemetry', (done) => {
+            var options: request.CoreOptions = {
+                qs: {
+                    deviceId: "SampleDevice001_648",
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: same as others, avoid hardcoding device id

File: apitests/spec/device_telemetry_spec.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        var deviceId: string;
+        beforeAll(function (done) {
+            req = request.defaults({ json: true, baseUrl: 'https://localhost:44305/api/v1/telemetry' });
+            done();
+        });
+
+        it('1. Get alert history', (done) => {
+            req.get('alertHistory', (err, resp, result: AlertHistory) => {
+                expect(result).toBeTruthy();
+                expect(result.data).toBeTruthy();
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: check for data.length > 0

File: apitests/spec/device_telemetry_spec.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result.devices[0].status).not.toBeNull();
+                done();
+            });
+        }, 10000);
+
+        it('2. Get Device Location Data', (done) => {
+            req.get('deviceLocationData', (err, resp, result: DeviceLocationData) => {
+                expect(result).toBeTruthy();
+                expect(result.deviceLocationList).toBeTruthy();
+                expect(result.maximumLongitude).toBeTruthy();
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: add deviceLocationList.length > 0

File: apitests/spec/device_telemetry_spec.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        beforeAll(function (done) {
+            req = request.defaults({ json: true, baseUrl: 'https://localhost:44305/api/v1/telemetry' });
+            done();
+        });
+
+        it('1. Get alert history', (done) => {
+            req.get('alertHistory', (err, resp, result: AlertHistory) => {
+                expect(result).toBeTruthy();
+                expect(result.data).toBeTruthy();
+                expect(result.devices).toBeTruthy();
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: devices.length > 0

File: apitests/spec/device_telemetry_spec.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+                expect(result.data[0].value).toBeTruthy();
+
+                expect(result.devices[0]).toBeTruthy();
+                expect(result.devices[0].deviceId).toBeTruthy();
+                expect(result.devices[0].latitude).toBeTruthy();
+                expect(result.devices[0].longitude).toBeTruthy();
+                expect(result.devices[0].status).not.toBeNull();
+                done();
+            });
+        }, 10000);
```
Comment (kapil-ms) [inline apitests/spec/device_telemetry_spec.ts]: why do we require a timeout here ?

Comment (handeadiguzel) [inline apitests/spec/device_telemetry_spec.ts]: Timeout is by default 5 secs for Jasmine tests, I had to increase it for some tests. Once we add retrying we could get rid of them.

File: apitests/spec/device_telemetry_spec.ts (hunk 4 above)
Comment (handeadiguzel) [inline apitests/spec/device_telemetry_spec.ts]: added a TODO for it

**Pull request merged**